### PR TITLE
delete popup added for entity delete action when user delete entity by unregister entity

### PR DESCRIPTION
**Pull request by AmbrishRamachandiran**
## Hey, I just made a Pull Request!

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx [hunk 1]
```diff
@@ -85,7 +85,21 @@ const Contents = ({
         setBusy(true);
         try {
           await state.deleteEntity();
+          const entityArray = [entity].flat();
```
Comment (jhaals) [inline plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx:88]: No need for making this into an array

File: plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx [hunk 2]
```diff
@@ -85,7 +85,21 @@ const Contents = ({
...
+          const entityNames = entityArray.map(
+            item => item.metadata.title ?? item.metadata.name,
+          );
           onConfirm();
+          const message =
+            entityNames.length === 1
+              ? `Removed entity '${entityNames[0]}'`
```
Comment (jhaals) [inline plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx:95]: We can probably skip the `'` quotes for the message :)

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the contribution, it would be great to also add a test for this change!

**Comment by AmbrishRamachandiran** [comment]
> Thanks for the contribution, it would be great to also add a test for this change!

@jhaals can you please review the changes 

Is the test case is must ? i have did couple of similar contribution where i didn't added test cases please check [19219](https://github.com/backstage/backstage/pull/19219) 
[18782](https://github.com/backstage/backstage/pull/18782)

File: plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx [hunk 3]
```diff
@@ -85,7 +85,20 @@ const Contents = ({
         setBusy(true);
         try {
           await state.deleteEntity();
+          const entityNames = [entity]
+            .flat()
+            .map(item => item.metadata.title ?? item.metadata.name);
```
Comment (jhaals) [inline plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx:90]: It's not necessary to create an array for a single item as you only operate on one item in this case.

**Review by jhaals — CHANGES_REQUESTED** [review]
There is already a test case in `UnregisterEntityDialog.test.tsx` for deletion so adding an assert that the `post` method was called for the alertApi is desired.

File: plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx (hunk 3 above)
Comment (AmbrishRamachandiran) [reply plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx:90]: Sure @jhaals  ! but why i added it as array is for the reason check [here](https://github.com/backstage/backstage/pull/19219/files/6c06d5a087670737adcf8adafc9f038cf2363570#r1292287232)

if that is not applicable here will remove the array please confirm

Comment (jhaals) [reply plugins/catalog-react/src/components/UnregisterEntityDialog/UnregisterEntityDialog.tsx:90]: I believe it's different from the table example if you look at the signature of the function. Line 52 indicates that entity is not an array so let's proceed with that assumption.

**Review by jhaals — CHANGES_REQUESTED** [review]

**Comment by AmbrishRamachandiran** [comment]
@jhaals Added test case with suggested changes please review

**Review by jhaals — APPROVED** [review]
Looking good, thanks!

**Pull request merged**